astro/src/content/docs/get-started/run-in-the-cloud/github-actions.mdx

```diff
@@ -0,0 +1,456 @@
+---
+title: GitHub Actions With FusionAuth
+description: How to use CI/CD with FusionAuth in GitHub.
+section: get started
+subcategory: run in the cloud
+prerequisites: Docker version 20
+url: https://github.com/FusionAuth/fusionauth_github_actions
+codeRoot: https://raw.githubusercontent.com/FusionAuth/fusionauth_github_actions/main
+---
+import InlineUIElement from 'src/components/InlineUIElement.astro';
+import Aside from '/src/components/Aside.astro';
+import {RemoteCode} from '@fusionauth/astro-components';
+import AppDatabaseDiagram from 'src/diagrams/docs/get-started/run-in-cloud/github-actions/app-database.astro';
+import ActionWorkflowDiagram from 'src/diagrams/docs/get-started/run-in-cloud/github-actions/action-workflow.astro';
+
+## Introduction
+
+This article explains:
+- How to upgrade FusionAuth.
+- How to write automated tests of your application login with FusionAuth.
+- How to run these tests in a [GitHub Actions workflow](https://docs.github.com/en/actions) when pushing to the main branch.
+- How to automate deployment and testing on your live application.
+
+To follow this article, you need to know how to create a project in GitHub, but you don't need to know anything about FusionAuth, GitHub Actions, or automated testing, as the basics are explained here.
+
+You will need Docker and Node.js installed to run the example application accompanying this guide.
+
+## Some Definitions And Goals
+
+FusionAuth provides login functionality for your web application. Users are stored in the FusionAuth database, separate from the database your app uses to keep its data. Assuming you have built a login page that redirects a user to FusionAuth, your system will have the components shown below.
+
+<AppDatabaseDiagram/>
+
+You might have the following questions:
+- How do I deploy these components to production?
+- What happens when FusionAuth releases a new version? How do I upgrade FusionAuth without losing data?
+- How can I test that login still works after upgrading my app or FusionAuth?
+- How do I separate and manage my development and production environments?
+- Where do I keep database passwords safely and include them when deploying?
+
+Automating these tasks is called CI/CD or continuous integration and continuous deployment.
+- CI refers to testing changes to code you push to the main branch to ensure it fits in with the existing system and everything still works.
+- CD refers to deploying every push to the main branch to your production (live) site.
+
+GitHub provides a tool for automating workflows called GitHub Actions. It provides a virtual machine in which you can run scripts to check out your code, compile it, deploy alongside FusionAuth and other services, run tests, and manage deployment.
+
+The flow looks like the diagram below.
+
+<ActionWorkflowDiagram/>
+
+## A Simple Example Using GitHub Actions
```

Comment:
@rideam I was just thinking about an image that would diagram what sits where because there are so many moving pieces. It is just a suggestion. If you do not think it is worth it, then just resolve this.

![Screenshot 2024-02-22 at 3 55 10 PM](https://github.com/FusionAuth/fusionauth-site/assets/137117976/935b39eb-03ce-4110-b165-332246471357)